### PR TITLE
Feat/multiple languages support

### PR DESCRIPTION
**Pull request by chiefEbube**
# Description
Implement the UI for multiple language support (English, French, Spanish) within the BOILERPLATE. This involves integrating a language selection feature into the navigation bar for quick access and ensuring that users can also change their preferred language via the settings page. The selected language should be consistently applied across all frontend components, and the interface must be fully mobile responsive.
https://linear.app/bingo-boiler/issue/BOI2-1/boilerplate-fe

**Closes #issue_number_here**

# Changes proposed

## What were you told to do?
- **Implement Language Selection Interface:**

Implement the UI for a language selection feature within the navigation bar, ensuring it is easily accessible and allows users to quickly switch between EN, FR, and ES.

Develop a secondary language selection interface on the settings page, providing an alternative option for users to change their preferred language.

Ensure both interfaces are intuitive and maintain consistency with the overall BOILERPLATE design.

- **Consistent Language Application:**

Implement a localization library (e.g., i18n, react-intl) to manage and apply translations across all frontend components.

Ensure that all text elements, including headings, labels, buttons, and notifications, reflect the selected language throughout the application.

Implement fallback mechanisms to handle cases where translations may be missing or incomplete.

- **Mobile Responsiveness:**

Ensure the language selection interfaces and all UI elements are fully mobile responsive, maintaining usability and visual consistency across various screen sizes and orientations.

Implement responsive design practices to ensure the interface adapts smoothly when switching languages or resizing screens.

+ **Documentation and Best Practices:**

Research available documentation on best practices for implementing multilingual support in web applications.

Follow coding standards and best practices for internationalization and localization to ensure maintainability and scalability.

## What did you do?

- This PR is a successful attempt to fulfill the above requirements.
# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos
Screenshot from 2024-08-13 18-52-37
Screenshot from 2024-08-13 18-51-47

### CODE REVIEW

**Comment by prudentbird** [comment]
Your commit messages are wrong

You have lint errors

File: public/images/flags/en.svg
Comment (prudentbird) [inline public/images/flags/en.svg:1]: Use an icon library!

File: messages/fr.json
Comment (prudentbird) [inline messages/fr.json:1]: Why supporting only 3 languages?

File: src/i18n.ts
Comment (prudentbird) [inline src/i18n.ts:1]: This file name isn't valid!

**Review by prudentbird — CHANGES_REQUESTED** [review]

File: messages/fr.json
Comment (chiefEbube) [reply messages/fr.json:1]: I'm supporting 3 languages because that's exactly what is in the linear ticket.

File: src/i18n.ts
Comment (chiefEbube) [reply src/i18n.ts:1]: Then this file name is coming from the library. It's not a custom name I just gave to the file.

File: public/images/flags/en.svg
Comment (chiefEbube) [reply public/images/flags/en.svg:1]: I'll check if this is possible and fix it.

**Comment by chiefEbube** [comment]
@PrudentBird, I have resolved the lint issues. But I can't use icons for the flags because lucide-react doesn't have that and also, I can't change i18n.ts file name because it comes with the next-int'l library

**Review by prudentbird — DISMISSED** [review]

**Review by prudentbird — APPROVED** [review]

**Review by phoenixdahdev — APPROVED** [review]

**Pull request merged**